beacon-chain/blockchain/process_block.go

```diff
@@ -41,109 +40,60 @@ const depositDeadline = 20 * time.Second
 // This defines size of the upper bound for initial sync block cache.
 var initialSyncBlockCacheSize = uint64(2 * params.BeaconConfig().SlotsPerEpoch)
 
+// postBlockProcessConfig is a structure that contains the data needed to
+// process the beacon block after validating the state transition function
+type postBlockProcessConfig struct {
+	ctx            context.Context
+	signed         interfaces.ReadOnlySignedBeaconBlock
+	blockRoot      [32]byte
+	headRoot       [32]byte
```

Comment:
`fcuConfig` already contains `headRoot`, do you really need another head root field here?

Reply:
Nvm, looks like you need it for `getFCUArgsEarlyBlock`